src/components/ArticleManager.jsx

```diff
@@ -5,23 +5,84 @@ import { useSearchParams } from 'react-router-dom';
 
 export default function ArticleManager(){
     const [articleList, setArticleList] = useState([])
+    const [isLoading, setIsloading] = useState(true)
 
     const [searchParams, setSearchParams] = useSearchParams();
     const topicQuery = searchParams.get('topic');
     const authorQuery = searchParams.get('author')
+
+    const [direction, setDirection] = useState('desc')
+    const [sortField, setSortField] = useState("null")
+
+    console.log(sortField)
+    // Setting direction asc/deesc
+    const onDirectionChange = (e) =>{
+        setDirection(e.target.value)
+    }
+    const setDirectionOrder = (e) => {
+        onDirectionChange(e)
+        const newParams = new URLSearchParams(searchParams);
+        newParams.set('order', direction);
+        setSearchParams(newParams)
+    }
+
+    // sorting 
+    const sortBasedOnKey = (property) =>{
+        const array = [...articleList];
+        return array.sort((a, b)=>{
+            let x = a[property];
+            let y = b[property];
+            return y-x;
+        })  
+    }
+    const handleSortChange = (e) =>{
+        setSortField(e.target.value)
+        if(e.target.value !== "null"){
+            let sortedData = sortBasedOnKey(e.target.value);
+            setArticleList(sortedData)
+        }
+    }
+
     useEffect(()=>{
-        getAllArticles(topicQuery, authorQuery)
+        getAllArticles(topicQuery, authorQuery, direction)
         .then((response)=>{
             setArticleList(response.data.articles);
+            setIsloading(false)
         })
     }, [searchParams])
 
+    if (isLoading)  return <p className="loading">Loading ... </p>
     return(
-        <section className="allArticles">
+        <main>
+            <div className="sortnavbar">
+                <div className="directionWrapper">
+                    <h3>Show articles:</h3>
+                    <div className="directionInputsWrapper">
+                    <label  htmlFor="asc">From new to old</label>
+                    <input className="radio" type="radio" name="direction" value="desc" id="desc" checked={direction === 'desc'} onChange={setDirectionOrder}/>
+
+                    <label className="labelDirection" htmlFor="asc">From old to new</label>
```

Comment:
This label of "old to new" is correct if they're sorting by date but doesn't quite make sense when they are sorting by ascending or descending by votes for example